packages/eas-cli/src/utils/easJson/EasJsonAccessorWithCustomErrors.ts

```diff
@@ -0,0 +1,38 @@
+import { EasJson, EasJsonAccessor } from '@expo/eas-json';
+import { InvalidEasJsonError } from '@expo/eas-json/build/errors';
+import chalk from 'chalk';
+import { ValidationError } from 'joi';
+
+import Log, { link } from '../../log';
+
+export const easJsonCustomErrorMessageHandlers: ((err: ValidationError) => void)[] = [
+  // Ask user to upgrade eas-cli version or check the docs when image is invalid.
+  (err: ValidationError) => {
+    for (const detail of err.details) {
+      if (detail.path[detail.path.length - 1] === 'image') {
+        Log.error(
+          `The image you specified in ${chalk.bold(
+            'eas.json'
+          )} is invalid for this EAS CLI version. Please check if your EAS CLI version is up to date. Othervise visit ${link(
+            'https://docs.expo.dev/build-reference/infrastructure/'
+          )} to learn more about supported images.`
+        );
+      }
+    }
+  },
+];
+
+export class EasJsonAccessorWithCustomErrorMessages extends EasJsonAccessor {
```

Comment:
I'm not strongly against it, but if possible I would prefer to avoid it if possible. The main problem with your first version was logging in eas-json package. Alternatively, you could:
- Replace the log with an exception that has a more specific error message,
- If different logging/formatting is necessary and you need to use a logger, I would suggest using a more generic name. `WithCustomErrorMessages` is the correct name, but if it's only the accessor we use, that information is redundant.
- It's a bit more work, but ideally, to combine the benefits of both solutions, you could move validation to eas-json, throw an exception there, and override how errors are handled in oclif so it's formatted in a way logger would do.

@dsokal do you have an opinion on that?

Reply:
No

Reply:
I think I will go with the first approach in this PR and create a separate task for completing the third approach which would be a good change overall.

Reply:
https://linear.app/expo/issue/ENG-6945/improve-eas-cli-error-logging